chore: introduce governance board (#1634)
asyncapi/community: CHARTER.md

---
<p align="center"><strong>Technical Charter (the “Charter”)</strong></p>
<p align="center">
  for</br>
  AsyncAPI Project a Series of LF Projects, LLC</br>

>   Adopted 03-19-2021, Amended [TODO:set a date once agreed with LF]

Author: LF is super slow in responses, so we will eventually apply lazy consensus, when we agree and long term vote that change is needed, we will wait for them some time and then merge if no veto

### 2. Technical Steering Committee

1. The Technical Steering Committee (the “TSC”) will be responsible for the oversight of the open-source Project.
2. The TSC appoints the Governance Board as described in Section 3.
3. TSC membership is not time-limited. There’s no maximum size of TSC. The size is expected to vary in order to ensure adequate coverage of important areas of expertise, balanced with the ability to make decisions efficiently. The TSC must have at least three members.
4. No more than one-fourth of the TSC members may be affiliated with the same employer. If the removal or resignation of a TSC member, or a change of employment/affiliation by a TSC member, creates a situation where more than one-fourth of the TSC membership shares an employer, then the situation must be assessed by TSC. In any case, the number of TSC members affiliated with the same employer must not be higher than one-third (round half up number).

> 5. Voting members of the Technical Steering Committee (TSC) include committers of any Project's code repository and AsyncAPI Ambassadors. The TSC documents in the project’s GOVERNANCE file the method for determining voting members.

Author: From my perspective, a director, who knows GB is not taking power away from TSC - I think the only controversial part of the proposal is to let Ambassadors be part of TSC

correct me if I'm wrong

6. TSC members may resign anytime and also ask to be added to TSC list anytime, as long as the requirements are met.
7. Any meetings of the TSC are intended to be open to the public, recorded, and can be conducted electronically, via teleconference, or in person.
8. TSC projects generally will involve Contributors, Committers, and Ambassadors. The TSC may adopt or modify roles so long as the roles are documented in the Project’s GOVERNANCE file.
9. TSC members are expected to regularly participate in TSC voting.

> 10. In the case where an individual TSC member -- within any three-month period -- does not participate in TSC votes, the member shall be automatically removed from the TSC.

Reviewer: Are we going to create some automation around it 💪?

Author: this was in the rules before, not a new stuff. 

but yeah, we have https://github.com/asyncapi/community/blob/master/TSC_VOTING_OVERVIEW.md and we are able to determine if someone was active or not. Remove automation yes, but it is pretty sensitive topic to rely on automation fully, so automation should be rather focused on nitifying the board to check with TSC members what is the situation - this is what I do manually now 🤷🏼

    9. discussions, seeking consensus, and where necessary, voting on technical matters relating to the code base that affect multiple projects; and
    10. coordinating any marketing, events, or communications regarding the Project.

### 3. Governance Board

1. The Project Governance Board (the “Board”) will provide oversight and guidance on strategic and operational matters, but will not replace the TSC in making decisions, unless the TSC delegates some responsibilities to the Board. Exceptions will be specified in the GOVERNANCE document.

> 2. The Board must have five members.

Author: we would start with 3, and elect another 2 next year - 5 is a target to reach, not initial state - this is the idea

> 3. The Board initially comprises three members and expands to five members after the first 12 months.

Reviewer 2: I like putting it in the GOVERNANCE file.

4. The Board member is elected for 24-month cadence.

> 5. The Board member is also a TSC member with all it's rights.

Reviewer 2: Then let me suggest the following:

```suggestion
5. The Board members MUST be TSC members with all their rights.
```

> 6. The Board will follow a 12-month election cycle, with members eligible for re-election only once. 

Reviewer 2: Ahh, I see. Good idea 👍

> 6. The Board will follow a 12-month election cycle, with members eligible for re-election only once. 

Reviewer 3: Shouldn't election be every 2 years? Because 12 months is too small to implement any changes honestly

Author: sorry, when I was replying in batch, I did not see your comments, please look at my answers to Fran - I think these answer your concerns

7. The Board elections will not be managed as regular TSC votes. Details of the voting process will be specified in the GOVERNANCE document.
8. A Board member can only be removed upon resignation or a majority of the TSC votes. New Board member election must be organized within 60 days.

> 9. The Board will appoint a chairperson from within its members every 12 months. The chairperson can only be replaced upon resignation or the majority of the Board members votes.

Reviewer 3: Mmmmh I'm confused here, so the chairperson will sit in the position for only 12 months and rotate ? Then what about the main GB election?

Author: yeah, being a chair might be overwhelming, so having a 12m window is good I think. Person can always be appointed again. Only Board members decide here, so it is different than normal elections

I will probably move this point to GOVERNANCE doc 🤔

10. The Board chairperson coordinates work of the board and ensures it operates effectively.

> 11. The Board is responsible for presenting votes to the TSC. The Board is preparing and managing high-sensitivity issues before the vote especially in cases involving sponsorships, external partnerships, or policy considerations. 

Reviewer 2: ```suggestion
11. The Board is responsible for presenting votes to the TSC. The Board must prepare and manage high-sensitivity issues before the vote, especially in cases involving sponsorships, external partnerships, or policy considerations. 
```

> 12. The Board assures effective voting process, participation and vote conclusion. 

Reviewer 2: ```suggestion
12. The Board must assure efficient voting processes, participation and vote conclusions.
```

> 13. Board members are expected to hold private monthly meetings under `Chatham House Rule` guidelines, with meeting summaries made available publicly.

Reviewer 2: Should we put a time limit in which these summaries must be available?

Reviewer 4: Worth to mention, Imho, the expectations of such a summaries. For example, and not limited to, a link to the recording + the transcription.

Reviewer 2: There can't be any recording or transcription because the meetings are private (see https://www.chathamhouse.org/about-us/chatham-house-rule). Only the outcome is public, to ensure people can speak freely without fearing consequences or we don't disclaim private data from companies/sponsors.

Reviewer 2: 3 days sounds good. Maybe we can be more specific and mention something like "3 days (excluding Saturdays and Sundays)"?

Author: 3 working days ?

> 14. Details on Board roles, duties, and operational processes are documented in a separate GOVERNANCE document for clarity and flexibility, covering:

Reviewer 2: ```suggestion
14. Details on Board roles, duties, and operational processes must be documented in a separate GOVERNANCE document for clarity and flexibility. They should cover:
```

    - Management of finances to ensure sustainability;

>     - Engagement with external companies and sponsorship;

Reviewer 2: ```suggestion
    - Engagement with external companies and sponsorships;
```

5. A Committer may be removed from a Project’s repository by the majority approval of the other Committers of the same repository.
6. Participation in the Project through becoming a Contributor and Committer is open to anyone so long as they abide by the terms of this Charter. 

### 5. TSC Voting

1. While the Project aims to operate as a consensus-based community, if any TSC decision requires a vote to move the Project forward, the voting members of the TSC will vote on a one vote per voting member basis.
2. Quorum requirements and voting rules will be specified in the GOVERNANCE document. If quorum is not reached after four weeks, the vote will be canceled, and the Board will reassess the topic.
3. Except as provided in Section 6, decisions require a majority of votes.
4. Decisions must be always made offline, by electronic vote.

> 5. The voting process is described in the GOVERNANCE. In any case, TSC members must have no more and no less than seven calendar days to vote.

Reviewer 2: ```suggestion
5. The voting process is described in the GOVERNANCE file. In any case, TSC members must have no more and no less than seven calendar days to vote.
```
